### ISSUE TITLE
Single-strand support, metaDMG output integration, Output discussion

### ISSUE THREAD

**Issue by RocesV** [LouisPwr/CarpeDeam#1]
Hi @LouisPwr @AnnSeidel ,

First of all, thank you very much for introducing this new software. I genuinely believe it has the potential to make a significant impact in the field! 😄 That said, I am relatively new to this area, so I apologize if any of my questions below seem straightforward.

###  **TASK:**  

I am working on assembling ancient single-stranded, single-end DNA reads from a coprolite sample. For some downstream analyses, I am particularly interested in generating the longest possible contigs.

I have performed metaDMG analysis (forward-only option) on a collection of samples, including this one, and confirmed ancient evidence for several taxa. After pre-processing, this sample contains approximately 45 million reads. As expected with this type of data, most of the reads are very short.

### **Q1:  Single-strand support**

Initially, when using CarpeDeam, I was unsure how to compute the damage matrices it requires. To experiment, I used DamageProfiler for five ancient taxa (authenticated through metaDMG) and calculated mean values to generate the damage matrices. Following the README instructions, I focused on C-to-T and G-to-A substitutions, setting all other substitutions to 0.

Here are the resulting matrices:

dhigh5p.prof
A>C     A>G     A>T     C>A     C>G     C>T     G>A     G>C     G>T     T>A     T>C     T>G
0       0       0       0       0       0.452558        0       0       0       0       0       0
0       0       0       0       0       0.232782        0       0       0       0       0       0
0       0       0       0       0       0.116812        0       0       0       0       0       0
0       0       0       0       0       0.104225        0       0       0       0       0       0
0       0       0       0       0       0.065798        0       0       0       0       0       0

dhigh3p.prof
A>C     A>G     A>T     C>A     C>G     C>T     G>A     G>C     G>T     T>A     T>C     T>G
0       0       0       0       0       0       0.030549        0       0       0       0       0
0       0       0       0       0       0       0.029873        0       0       0       0       0
0       0       0       0       0       0       0.023769        0       0       0       0       0
0       0       0       0       0       0       0.021263        0       0       0       0       0
0       0       0       0       0       0       0.025229        0       0       0       0       0

Given that my data is single-stranded, my damage matrices look like this. Is CarpeDeam expected to work well under these conditions? If not, is single-strand support planned for future releases?

### **Q2: metaDMG output integration**

Since @genomewalker (involved in metaDMG) is part of this project, I was wondering if any metaDMG output files could serve as a proxy for the required damage matrices, thereby eliminating the need for DamageProfiler. For example, could files like *.mismatches.txt or, in my case (using the forward-only option), the "f_CT" column from mismatches.parquet be used?

Given metaDMG's growing adoption in the community, a brief tutorial or guide on parsing metaDMG outputs into CarpeDeam-compatible damage matrices would be extremely helpful for future users.

### **Q3: Output discussion**

The speed of CarpeDeam is impressive! Using the following command:

`carpedeam ancient_assemble $INPUT/$FASTQ $FASTAOUTPUT $TMPDIR --ancient-damage $INPUT/dhigh --threads 90 --unsafe`

For this sample (~45M reads), I obtained 43 contigs, with the longest being ~1,240 bp. This result was surprising as I initially expected more and longer contigs. I understand that many factors could improve the assembly—e.g., deeper sequencing or filtering reads for only those higher taxonomic ranks I am interested in —but I noticed other studies using comparable samples often retrieved longer and more numerous contigs without employing a damage-aware assembler. For example:

- Table S2 in https://environmentalmicrobiome.biomedcentral.com/articles/10.1186/s40793-024-00644-8
- Table S1c and S2a in https://www.science.org/doi/10.1126/science.adf5300

Although these were preliminary runs, I would greatly appreciate any insights you might have on these results.

Once again, thank you for the excitement and innovation this tool has brought to the field! 😉 

Cheers,

Víctor

**Comment by LouisPwr** [comment]
Hi Victor,

Thank you for your interest in our work, and apologies for the delayed response - your issue unfortunately got buried in my list.

To address your questions:

Q1: We haven’t explicitly tested the tool on single-stranded libraries, but in principle, it should work as expected. However I cannot predict how exactly it would benchmark against other assemblers. 

Q2: That's a great suggestion and could indeed be valuable. However, implementing this would require significant changes to the core database structure, so it's not something planned in the immediate future.

Q3: Those are indeed quite low numbers. Could it be that the reads are not adapter-trimmed? I once had a similar situation where a user thought the adapter trimming takes automatically place (as in SPAdes). However, it is not built-in in CarpeDeam. Also, since you're working with single-end data, reads can't be merged, which results in shorter input sequences, which can be another factor that might contribute to reduced performance (Since merging does increase the fragment length definitely by a bit). In general, we've observed that assembler performance can vary significantly depending on the metagenomic dataset, so it's often hard to generalize across different contexts.

Again, sorry for the delay!
Louis

**Comment by Louis-MG** [comment]
Hello ! I would like to follow up on the output discussion. I was surprised by the low number of contigs as well, before realising that the minimum length in the output is set to 500bp (seqkit stats helped a bit). I would be interested in knowing the rational behind it ? @RocesV Could it be an explenation of the low number of contigs you observed ?
Another question: PenguiN is an assembler that was "specially tested" for viral DNA and microbial 16S genes, which is why I guess you targeted tRNAs and rRNAs in your benchmark. Do you expect/did you observe a bias toward these sequences in the assemblies of CarpeDeam ?

**Comment by LouisPwr** [comment]
Hi Louis,
Thanks for trying CarpeDeam.

Regarding the number of contigs: Did you trim adapters and merged the reads prior to assembly? 
Additionally, dataset-specific factors such as sequencing depth can greatly influence the number and length of resulting contigs.

You can adjust the minimum contig length using the` --min-contig-len` parameter. By default, it is set to 500 bp to avoid overly bloated output files filled with short, often uninformative sequences that could not be assembled. However, this threshold is somewhat arbitrary and it also depends on your research objectives.

And about the RNA sequences: There is a bias in a sense that CarpeDeam seems to do a good job in recovering such sequences (see the latest version of the paper on bioRxiv, we included a 16s-rRNA analysis). 
Currently, I would say CarpeDeam is best suited for datasets where protein prediction or 16S rRNA analysis is the primary goal. For full genome reconstruction of MAGs, the output can still be too redundant, and for some datasets the assembler may struggle with assembling ultra-long sequences.

Let me know how it goes.

**Comment by Louis-MG** [comment]
Yes I did merge the reads using leeHom as recommended. Thank you for the answers and clarifications.
Regarding Victor's Q2, I had to make a bash script to convert metaDMG print output to the correct format, if that can help ! I do agree that moving some information to a more detailed wiki would be best.

Have a nice day !
